Construção da pipeline de CI do packer

hazkall opened the pull request:
# Titulo

<!-- Por favor descreva seu pull request aqui. -->

- [x] Garanta que seu **topic/feature/bugfix branch** tenha uma branch nomeada e não a sua branch main esteja no PR
- [x] Dê um titulo que expresse o objetivo do PR
- [x] Associe seu PR a uma Issue criada no repositósito. Caso seja uma correção de linguagem ou pequenas correções, não é necessário
- [x] Descreva o objetivo do PR
- [x] Inclua links relevantes para a sua modificação/sugestão/correção
- [ ] Descreva um passo-a-passo para testar o seu PR

## Issue

close #5
close #6 
close #9 

## Objetivo

Esse PR tem como objetivo a criação da pipeline de construção da imagem do packer no momento do release


## Referências

<https://github.com/lipenodias/iac-packer-orquestradores/actions>

## Como testar

<!--
Marque um `x` dentro de [ ] para os itens que você forneceu informação
Para modificar este template no seu repositório, basta criar o arquivo .github/pull_request_template.md nele.
-->

@lgfa29 @snifbr @marcelomansur @thasz @nerdtux @marlensouza

nerdtux committed:
ci: Adicionando arquivo ci.yml para construçao de CI.
Utilizamos template disponibilizado no GGitHub da Hashicorp para CI do Packer.
Setamos SECRETS com as credenciais da GCP.

Revisado por:
@lipenodias
@thasz
@marcio-machado76
@jremerich

Ref: issue #6

Commit:
ci: Construção da pipeline de CI do packer

Commit com a construção do processo de CI da imagem do packer
Feito também o acrescimento no requirements.yml a documentação do
ansible galaxy nos comentarios
Corrigido a variavel de ambiente do Projeto ID, ela deve ser buscada
exatamente igual ao que esta declarado dentro de env(GCP_PROJECT).
Function do HCL env() usada em variables.pkr.hcl.
Adicionado o step de instalação do ansible e packer, visto que o actions default
da hashicorp não possui ansible.

Revisores:
- Gabriel Inglesias
- hmljunior
- Luiz Aoqui
- Marlen_Souza
- plinio
- Thais d' Angelis
- Vinicius-Cursi

Ref: #6

Commit:
ci: Construção da pipeline de CI do packer

Commit com a construção do processo de CI da imagem do packer
Feito também o acrescimento no requirements.yml a documentação do
ansible galaxy nos comentarios
Corrigido a variavel de ambiente do Projeto ID, ela deve ser buscada
exatamente igual ao que esta declarado dentro de env(GCP_PROJECT).
Function do HCL env() usada em variables.pkr.hcl.
Adicionado o step de instalação do ansible e packer, visto que o actions default
da hashicorp não possui ansible.

Revisores:
- Gabriel Inglesias
- hmljunior
- Luiz Aoqui
- Marlen_Souza
- plinio
- Thais d' Angelis
- Vinicius-Cursi

Ref: #6

hazkall committed:
Merge pull request #2 from lipenodias/dev

Construção da pipeline de CI do packer

gomex reviewed (commented):
Seria legal ter esse pipeline rodando completo no fork, assim podemos ver os possíveis problemas. Não me parece que tem nenhum problema nesse código, mas é sempre bom testar.

Commit:
feat(ci): Adiciona tag da release no nome da image

Conforme solicitado no projeto foi adicionado ao nome da imagem do
packer a tag de release do github.
Caso não seja passada uma tag de release será adicionado o timestamp no
lugar.

Foi também alterado as roles do ansible para as versões de release mais
atuais, já suportando a dependencia da nova role iac_role_runtime.

Revisor por:
- Danilo Rocha
- Luiz Aoqui
- Thais d'Angelis
- Karollyne Costa
- Mansur
- Raph

Ref: #5 #6 #9

hazkall committed:
Merge branch 'main' into dev

hazkall committed:
Merge pull request #3 from lipenodias/dev

teste

Commit:
fix: Correção da condição do image tag

Foi necessaria a correção do ternario de condição que estava recebendo
sempre o valor null, quebrando o build da imagem.

Revisor por:
- Danilo Rocha
- Luiz Aoqui
- Thais d'Angelis
- Karollyne Costa
- Mansur
- Raph

Ref: mentoriaiac#5 mentoriaiac#6 mentoriaiac#9

hazkall committed:
Merge pull request #4 from lipenodias/dev

fix: Correção da condição do image tag

Commit:
fix: Foi necessário corrigir nome da imagem

Devido a regras da gcp foi necessário fazer um replace no nome da imagem
para remover os pontos gerados pela tag de release.

Revisor por:
- Danilo Rocha
- Luiz Aoqui
- Thais d'Angelis
- Karollyne Costa
- Mansur
- Raph

Ref: mentoriaiac#5 mentoriaiac#6 mentoriaiac#9

Commit:
fix: Foi necessário corrigir nome da imagem

Devido a regras da gcp foi necessário fazer um replace no nome da imagem
para remover os pontos gerados pela tag de release.

Revisor por:
- Danilo Rocha
- Luiz Aoqui
- Thais d'Angelis
- Karollyne Costa
- Mansur
- Raph

Ref: mentoriaiac#5 mentoriaiac#6 mentoriaiac#9

hazkall committed:
Merge pull request #5 from lipenodias/dev

fix: Foi necessário corrigir nome da imagem

hazkall committed:
Merge branch 'main' into dev

hazkall committed:
Merge pull request #6 from lipenodias/dev

fix: Foi necessário corrigir nome da imagem

lipenodias commented:
@gomex 

Realizado o teste com a criação da imagem com a release sendo incorporada no nome da imagem.


![image](https://user-images.githubusercontent.com/64644915/136715171-e12c5718-de3a-49c5-a3a2-b1cfb65cad42.png)


![image](https://user-images.githubusercontent.com/64644915/136715190-264b6c14-c848-4a32-a713-8ca36fc4a877.png)

snifbr committed:
rm: arquivo fora do escopo do PR

snifbr committed:
fix: arquivo fora do escopo do PR

snifbr committed:
fix: arquivo fora do escopo do PR

snifbr commented:
LGTM

gomex commented:
Muito bom!!! Eu to AFK agora, mas reviso ao menos até amanhã.
